### PR TITLE
Fix a few bugs with top level schema definitions

### PR DESCRIPTION
**Pull request by acao**
- [x] a few tests to demonstrate a few failing test cases
- [ ] fix the top level $ref bug
- [ ] fix the top level `allOf` bug #73  and #69

### CODE REVIEW

File: src/__tests__/json-completion.spec.ts [hunk 1]
```diff
@@ -291,6 +292,57 @@ describe("jsonCompletion", () => {
       },
     ]);
   });
+  it("should autocomplete for array of objects with filter", async () => {
+    await expectCompletion('{ "arrayOfObjects": [ { "f|" } ] }', [
```
Comment (imolorhe) [inline src/__tests__/json-completion.spec.ts:296]: this works for me locally

File: src/__tests__/json-completion.spec.ts [hunk 2]
```diff
@@ -291,6 +292,57 @@ describe("jsonCompletion", () => {
...
+        detail: "string",
+        info: "",
+        label: "foo",
+        template: '"foo": "#{}"',
+        type: "property",
+      },
+    ]);
+  });
+  it("should autocomplete for a schema with top level $ref", async () => {
+    await expectCompletion(
```
Comment (imolorhe) [inline src/__tests__/json-completion.spec.ts:307]: working on this one ⏱️

File: src/__tests__/json-completion.spec.ts [hunk 3]
```diff
@@ -291,6 +292,57 @@ describe("jsonCompletion", () => {
...
+          type: "property",
+          detail: "number",
+          info: "",
+          label: "bar",
+        },
+      ],
+      { schema: testSchema3 }
+    );
+  });
+  it("should autocomplete for a schema with top level complex type", async () => {
```
Comment (imolorhe) [inline src/__tests__/json-completion.spec.ts:326]: working on this one ⏱️

**Comment by acao** [comment]
@imolorhe from what i can tell it is because our getPointer method skips the root

**Comment by imolorhe** [comment]
@acao I was able to get it working without touching getPointer. Just touching the completion logic, and all tests are passing!

**Comment by imolorhe** [comment]
Since I'm on it, want to tackle a few other test cases before creating a PR

**Comment by acao** [comment]
@imolorhe awesome work! yeah I saw that you added these test cases and todos for the ones left. was awesome to see so many more standard schemas working! closing this for now as the next steps are already in the TODO comments

**Pull request closed without merging**